internal/command/deploy/deploy.go

```diff
@@ -177,10 +177,15 @@ func DeployWithConfig(ctx context.Context, appConfig *appconfig.Config, args Dep
 	return err
 }
 
-func deployToMachines(ctx context.Context, appConfig *appconfig.Config, appCompact *api.AppCompact, img *imgsrc.DeploymentImage) error {
+func deployToMachines(ctx context.Context, appConfig *appconfig.Config, appCompact *api.AppCompact, img *imgsrc.DeploymentImage) (err error) {
 	// It's important to push appConfig into context because MachineDeployment will fetch it from there
 	ctx = appconfig.WithConfig(ctx, appConfig)
 
+	metrics.Started(ctx, "deploy_machines")
+	defer func() {
+		metrics.Status(ctx, "deploy_machines", err == nil)
+	}()
+
 	md, err := NewMachineDeployment(ctx, MachineDeploymentArgs{
```

Comment:
what is the difference with 
```
defer metrics.Status(ctx, "deploy_machines", err == nil) 
```

Reply:
Is that the same?
I'd assume that, written like that, `err == nil` would be evaluated in-place, rather than at the end of scope, but I've never formally studied the language :)

Reply:
Based on this, I *think* what I did was right
![image](https://user-images.githubusercontent.com/13626644/234134962-fdd6e07a-1cb3-4231-9797-0c56126bee7f.png)

Reply:
indeed. you are right.